### ISSUE TITLE
summa.exe --version not matching the tag/hash installed

### ISSUE THREAD

**Issue by alexandermichels** [CH-Earth/summa#485]
I have tried to install SUMMA v3.0.3 using the tag and the specific commit linked to that tag, but everytime I do so, I end up with the following result from `summa.exe --version`

```
----------------------------------------------------------------------
     SUMMA - Structure for Unifying Multiple Modeling Alternatives
                           Version: v3.1.0
             Build Time: Fri 13 Aug 2021 01:35:09 PM CDT
                  Git Branch: tags/v3.0.3-0-g4ee457d
          Git Hash: 4ee457df3d3c0779696c6388c67962ba76736df9
----------------------------------------------------------------------
```

As you can see, the "Git Branch" says v3.0.3 and the hash also lists 4ee457d (as listed in the tagged version: https://github.com/NCAR/summa/releases/tag/v3.0.3), but the Version still reports v3.1.0.

Fortran version: GNU Fortran (GCC) 8.3.0
netCDF-Fortran: 4.5.2
Operating system: Ubuntu 20.04

Possibly related to: #367

**Comment by wknoben** [comment]
Hi Alexander, 

Thanks for reporting this. I think this can be fixed with a small change in the Makefile. Can you try replacing lines 302-306 in the Makefile with the following and report back?

```
VERSIONFILE = $(DRIVER_DIR)/summaversion.inc
BULTTIM = $(shell date)
GITBRCH = $(shell git describe --long --all --always | sed -e's/heads\///')
GITHASH = $(shell git rev-parse HEAD)
VERSION = $(shell git show-ref --tags | grep $GITHASH | sed 's/.*tags\///')
```

This changes how `VERSION` is determined and uses the `GITHASH` variable to do so, which should make sure you get the version that belongs to the tag. This will return nothing if the hash has no tag though, so I won't turn this into a pull request just yet.

**Comment by alexandermichels** [comment]
@wknoben Thank you for the reply! 

I manually edited the Makefile to the say the version I was installing. The diff for the edit was just:
```
--- build/Makefile.original	2021-08-19 12:56:30.191385895 -0500
+++ build/Makefile	2021-08-19 12:56:56.079512711 -0500
@@ -300,7 +300,7 @@
 
 # Define version number
 VERSIONFILE = $(DRIVER_DIR)/summaversion.inc
-VERSION = $(shell git tag | tail -n 1)
+VERSION = "v3.0.3"
 BULTTIM = $(shell date)
 GITBRCH = $(shell git describe --long --all --always | sed -e's/heads\///')
 GITHASH = $(shell git rev-parse HEAD)
```

I just wanted to make sure you were aware of the issue and it's good to know you already have a solution! I'll be sure to use this solution next time I have to install SUMMA!

**Comment by wknoben** [comment]
Addressed in #488. Closing this for now
